browser/android/background_video/background_video_playback_tab_helper.cc

```diff
@@ -19,17 +19,53 @@
 #include "url/gurl.h"
 
 namespace {
-constexpr char16_t kYoutubeBackgroundPlaybackScript[] =
-    u"(function() {"
-    "    if (document._addEventListener === undefined) {"
-    "        document._addEventListener = document.addEventListener;"
-    "        document.addEventListener = function(a,b,c) {"
-    "            if(a != 'visibilitychange') {"
-    "                document._addEventListener(a,b,c);"
-    "            }"
-    "        };"
-    "    }"
-    "}());";
+constexpr char16_t kYoutubeBackgroundPlaybackAndPipScript[] =
+    uR"(
+(function() {
+    if (document._addEventListener === undefined) {
+        document._addEventListener = document.addEventListener;
+        document.addEventListener = function(a,b,c) {
+            if(a != 'visibilitychange') {
```

Comment:
there should also be a browser test here, but there's a potential race condition here if the tab is backgrounded while it's still loading. We should only suppress this event if `document.hidden = false`.

Reply:
we should also have a test for this functionality in general, but both of these can be done in a follow-up

Reply:
Filed a new GitHub issue to make sure this does not fall though the cracks 👉 https://github.com/brave/brave-browser/issues/45668